Scenario_Duplication_Method.py

```diff
@@ -0,0 +1,30 @@
+class ScenarioManager:
+
+    def duplicate_scenario(self, original_scenario_id):
+        # Fetch the original scenario by ID
+        original_scenario = self.get_scenario_by_id(original_scenario_id)
+
+        # Create a copy of the scenario structure
+        new_scenario = Scenario()
+        new_scenario.name = f"Duplicate of {original_scenario.name}"
+        
+        # Iterate over tasks in the original scenario

```

Comment:
There are a lot more properties in a Scenario than just `tasks`

We also need to duplicate those

Reply:
I found this helpful